### ISSUE TITLE
Windows local development installation bugs

### ISSUE THREAD

**Issue by codegaze** [jekyll/jekyll-admin#175]
I know you are aware of these. I just wanted to be documented. If you'd like a PR for the documentation page is easy.
- `script` doesn't work for windows machines (I used the git bash as a workaround) so, to make it work:
  - bundle install (Ruby dependencies)
  - npm install (JS dependencies)
  - npm start

**Comment by benbalter** [comment]
@codegaze Yes! Thanks for opening this. For Appveyor (which now runs on each PR), and in my local testing in a VM, we rely on http://www.howtogeek.com/howto/41382/how-to-use-linux-commands-in-windows-with-cygwin/, which allows you to invoke `sh script/boostrap` and `sh script/test-server`. 

To note, the above bugs should affect local development, and should not running the compiled Gem. 

Any interest in putting some documentation together on how to contribute on Windows?

**Comment by codegaze** [comment]
No prob @benbalter, that was my intention, but first I must do something to make the API work on my local development machine to check everything is working as it should be. :confused:

**Comment by codegaze** [comment]
@benbalter Have you had any problems executing 'cibuild-ruby' from your VM using cygwin?

**Comment by benbalter** [comment]
Nope. You can [see it running](https://ci.appveyor.com/project/benbalter/jekyll-admin/branch/master/job/fj4il9es3mgw218f) on the latest build, and [our setup script](https://github.com/jekyll/jekyll-admin/blob/master/appveyor.yml#L9-L14). Where are you running into trouble?

**Comment by codegaze** [comment]
Some errors for reference:

**Error 1**
```
script/fmt:
 line 2: $'\r': command not found
: invalid option 3: set: -
```
This can be solved by running dos2unix "script" with each script.

**Error 2**
```
C:\Ruby22-x64\bin\ruby.exe: No such file or directory -- /cygdrive/c/Ruby22-x64/bin/bundle (LoadError)
```
This must be a setup issue, (more [here](https://github.com/oneclick/rubyinstaller/wiki/Troubleshooting#mix-and-matching-of-rubyinstaller-with-cygwin-will-fail)) if I run the script in the command line 'RACK_ENV=test JEKYLL_LOG_LEVEL=warn bundle.bat exec rspec' I get no error.

As I said earlier everything works fine with the git bash. Maybe this could be an "official" way to run the scripts?

**Comment by benbalter** [comment]
Ahh. Yes. I ran into the same issue I had to [add this line](https://github.com/jekyll/jekyll-admin/blob/master/appveyor.yml#L33) to the setup script to ensure git normalized the line endings from unix to windows.

**Comment by benbalter** [comment]
> As I said earlier everything works fine with the git bash. Maybe this could be an "official" way to run the scripts?

You know more than I do here. If you think that's the easier route, glad to call that the "official" way.

**Comment by mertkahyaoglu** [comment]
Since we merged #181, I'm closing this. Thanks again @codegaze ❤️
